xwiki-rendering-syntaxes/xwiki-rendering-syntax-xwiki20/src/main/java/org/xwiki/rendering/internal/renderer/xwiki20/reference/XWikiSyntaxResourceRenderer.java

```diff
@@ -139,17 +149,16 @@ public void endRenderLink(XWikiSyntaxEscapeWikiPrinter printer, ResourceReferenc
         printer.print(serialize(reference, freestanding));
 
         // If there were parameters specified, print them
-        printParameters(printer, reference, parameters);
+        printParameters(printer, parameters);
 
-        if (this.forceFullSyntax.peek() || !freestanding) {
+        if (this.forceFullSyntaxDeque.peek() || !freestanding) {
             printer.print("]]");
         }
 
-        this.forceFullSyntax.pop();
+        this.forceFullSyntaxDeque.pop();
     }
 
-    protected void printParameters(XWikiSyntaxEscapeWikiPrinter printer, ResourceReference resourceReference,
-        Map<String, String> parameters)
+    protected void printParameters(XWikiSyntaxEscapeWikiPrinter printer, Map<String, String> parameters)
```

Comment:
This one is actually a false positive. `org.xwiki.rendering.internal.renderer.xwiki21.reference.XWikiSyntaxResourceRenderer` override this method and use the passed `resourceReference`.